zookeeper-server/src/main/java/org/apache/zookeeper/server/Request.java

```diff
@@ -181,8 +182,7 @@ public byte[] getSerializeData() {
             try {
                 this.serializeData = Util.marshallTxnEntry(this.hdr, this.txn, this.txnDigest);
             } catch (IOException e) {
-                LOG.error("This really should be impossible.", e);
-                this.serializeData = new byte[32];
+                throw new UncheckedIOException(e);
```

Comment:
I'd prefer to include this change in another PR so that if it causes issues, we can confidently revert that one instead of coupled with other changes in this patch.

Since we have a lazy consensus on this change, I don't think another PR will block it's review and merge process.

Reply:
Fine. I will push another pr to fix it.